marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadHubArtifactsCommand.java

```diff
@@ -142,17 +142,28 @@ public void execute(CommandContext context) {
     }
 
     /**
-     * Per DHFPROD-2772, module permissions are being assigned to hub artifacts, which matches what is done for user
-     * artifacts. This ensures that in ML 10, each hub artifact has some permissions on it and can thus be accessed by
-     * a non-admin user when DHF is installed by a user without any default permissions (e.g. the admin user).
+     * As of 5.2.0, artifact permissions are separate from module permissions. If artifact permissions
+     * are not defined, then it falls back to using default permissions.
      *
-     * @param collections
+     * @param collection
+     * @param config
      * @return
      */
-    protected DocumentMetadataHandle buildDocumentMetadata(String... collections) {
+    protected DocumentMetadataHandle buildMetadata(HubConfig config, String collection) {
         DocumentMetadataHandle meta = new DocumentMetadataHandle();
-        meta.getCollections().addAll(collections);
-        documentPermissionsParser.parsePermissions(hubConfig.getModulePermissions(), meta.getPermissions());
+        String permissions = null;
+        switch (collection) {
+            case "http://marklogic.com/data-hub/flow":
+                permissions = config.getFlowPermissions();
+                break;
+            case "http://marklogic.com/data-hub/step-definition":
+                permissions = config.getStepDefinitionPermissions();
+                break;
+            default:
+                throw new RuntimeException("The provided collection name for the artifact is incorrect.");
```

Comment:
I don't think this is necessary. If neither is a match, it's a programming error - just return null. An automated test should verify that we get the expected permissions for flows and step definitions.